openupgrade_scripts/scripts/hr_expense/15.0.2.0/post-migration.py

```diff
@@ -10,6 +10,23 @@ def _fill_payment_state(env):
         WHERE account_move_id IS NULL
         """,
     )
+    # Starting from v15, credit journal items linked to expenses will be marked
+    # as excluded from invoice tab. Existing journal entries will be recomputed
+    # below, and this information should be properly set, otherwise amounts
+    # will be unset
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move_line
+        SET exclude_from_invoice_tab=(
+            CASE WHEN credit > 0.0 THEN true
+            WHEN debit > 0.0 THEN false
+            ELSE exclude_from_invoice_tab
+            END
+        )
+        WHERE expense_id IS NOT NULL
```

Comment:
For touching less rows:

```suggestion
        UPDATE account_move_line
        SET exclude_from_invoice_tab=true
        WHERE expense_id IS NOT NULL
        AND credit > 0
```

Reply:
Yes, that's correct, and I've verified that such account move lines have `exclude_from_invoice_tab` unset for a v14 database, indeed .

But I've found this problem: using OpenUpgrade with a database from v11 to v15, and every v14 account move line linked to expenses were migrated with `exclude_from_invoice_tab=true`, I still don't know why. That's why I've preferred this approach.

Anyway, if you think is more accurate your approach, no problem changing it.

Reply:
So is it a problem that both lines have `exclude_from_invoice_tab=true`?

And do you have `hr_expense_invoice` installed to know what happen to these items?

Reply:
> So is it a problem that both lines have `exclude_from_invoice_tab=true`?

Yes, according to

https://github.com/odoo/odoo/blob/15.0/addons/account/models/account_move.py#L1525-L1530 

for this kind of moves, total_amount is set to 0.0 if `exclude_from_invoice_tab` is the same value (no matter `true` or `false` because with `false` both balances are added and sum is `0.0`, and with `true` both are ignored), one of them must be `true` and the other `false`, as v15 new code indicates.

> And do you have `hr_expense_invoice` installed to know what happen to these items?

Not installed in origin. Should I install it?

Reply:
The `hr_expense_invoice` is to check if the journal entries linked to both the invoice and the expense are correctly kept, as for example the tax line should be excluded from invoice tab. I'm having now the doubt if your expense contains taxes, that this is the correct approach. Please check how it's done in v15.

Reply:
OK, I'll take a look at v15 implementation and will update. And in my case, every expense comes without taxes.

Thanks for all Pedro.